Data Structures: Size of the Set (#34506)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/08-coding-interview-prep/data-structures/size-of-the-set.english.md

## Description
<section id='description'>

>   Set class should also have a <code>size()</code> method which returns the size of the collection.

Reviewer: I would remove the word "also" here so the challenge is standalone.

Author: This is continuation of previous challenges in which the camper created "add" and "remove" methods.
Do you still think we should remove it?

Reviewer 2: If this is a continuation from the previous challenge, I don't see the harm in adding the word `also`.  My only suggestion is to start the sentence with `The set` instead of `Set`.

</section>

## Instructions
<section id='instructions'>

>   In this exercise we are going to create a <code>size</code> function for our Set. This function should be named <code>this.size</code> and it should return the size of the collection.

Reviewer 2: My suggestion is to write the instructions in second person and one other small change.
```suggestion
In this exercise you are going to create a <code>size</code> function for the `Set` class. This function should be named <code>this.size</code> and it should return the size of the collection.
```

```js
function Set() {
    var collection = [];
    this.has = function(element) {
        return (collection.indexOf(element) !== -1);
    };
  
    this.values = function() {
        return collection;
    };

    this.add = function(element) {
        if(!this.has(element)){
            collection.push(element);
            return true;
        }
        return false;
    };

    this.remove = function(element) {
        if(this.has(element)){
           var index = collection.indexOf(element);
            collection.splice(index,1);
            return true;
        }
        return false;
    };

    this.size = function() {
        return this.values().length;

>     }

Reviewer: semi
